### ISSUE TITLE
Elasticsearch: add raw query editor

### ISSUE THREAD

**Issue by gabor** [grafana/grafana#57410]
FIXME

(discussion at https://github.com/grafana/grafana/discussions/57409)

**Comment by Stono** [comment]
_please_.

**Comment by jade-lucas** [comment]
This would be very helpful indeed and provide full flexibility to someone that is experienced with elastic. I find the query builder limiting and prevents using useful feature that the search api providers. Such as leveraging runtime fields. Kibana is a good visual tool, but nothing beats grafana in terms of building visuals. A tight integration between elastic\opensearch and grafana would be a dream come true.

Elastic's search api
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html

**Comment by jprobbe** [comment]
Agreed. Actually I'm struggling with script_fields syntax. It seems supported by ES datasource, but reading the code to understand how it works is not obvious. It would be so easy to allow to write a JSON payload template in query editor, with variable substitution support.

**Comment by faxm0dem** [comment]
This feature existed before, and was then moved to a [discussion](https://github.com/grafana/grafana/discussions/57409) and then closed as "wontfix because too old". If you check out the above link, you can see there apparently was a community UX call that would discuss it. I haven't found the transcription, recorded video of this call, nor the conclusion.

I also believe the raw query needs to come back : while the current simplified query editor does the job in most cases, it's still frustrating not to be able to tackle some corner cases (and having to use kibana instead). 

See also https://github.com/grafana/grafana/issues/74142 which has interesting ideas

**Comment by idastambuk** [comment]
A design doc for this feature is currently in progress.

**Comment by bossinc** [comment]
https://github.com/grafana/grafana/pull/114066
